--- lib/Support/Unix/Program.inc.orig	2023-09-27 01:17:44.885372697 +0200
+++ lib/Support/Unix/Program.inc	2023-09-27 01:19:05.556546536 +0200
@@ -341,7 +341,14 @@
 namespace sys {
 
 #ifndef _AIX
+#ifdef EMSCRIPTEN
+pid_t wait4(pid_t pid, int* status, int options, struct rusage *usage) {
+  errno=ENOSYS;
+  return -1;
+}
+#else
 using ::wait4;
+#endif
 #else
 static pid_t(wait4)(pid_t pid, int *status, int options, struct rusage *usage);
 #endif
