--- lib/Support/Unix/Program.inc.orig	2024-08-18 21:13:21.487942069 +0200
+++ lib/Support/Unix/Program.inc	2024-08-19 09:19:29.571213211 +0200
@@ -345,8 +345,15 @@
 #if defined(_AIX)
 static pid_t(wait4)(pid_t pid, int *status, int options, struct rusage *usage);
 #elif !defined(__Fuchsia__)
+#ifdef EMSCRIPTEN
+pid_t wait4(pid_t pid, int* status, int options, struct rusage *usage) {
+  errno=ENOSYS;
+  return -1;
+}
+#else
 using ::wait4;
 #endif
+#endif
 
 } // namespace sys
 } // namespace llvm
